app/Http/Controllers/MgtController.php

```diff
@@ -229,6 +229,29 @@ public function getControllerMentor($cid, $facility = null) {
             return redirect("/mgt/controller/$cid")->with("success", "Successfully removed mentor role");
         }
     }
+    
+    public function getControllerInstructor($cid, $facility) {
+        if (!RoleHelper::isVATUSAStaff() && !RoleHelper::isFacilitySeniorStaff()) {
```

Comment:
isFacilitySeniorStaff should be checked against $facility in the request. Otherwise this checks if the requester is staff at their home facility, which might not be the same as $facility.